Sources/Fluent/FluentProvider.swift

```diff
@@ -13,7 +13,8 @@ extension Request {
                 id,
                 logger: self.logger,
                 on: self.eventLoop,
-                history: self.fluent.history.historyEnabled ? self.fluent.history.history : nil
+                history: self.fluent.history.historyEnabled ? self.fluent.history.history : nil,
+                maxPerPage: self.fluent.pagination.maxPerPage
```

Comment:
I notice know how what I suggested might conflict with another comment I wanted to make. If you only specify the maxPerPage on the application level and not on the request, it should default to the one on application. You'd have to "merge" the structs.
Now that I'm saying this I'm also realizing that the two Pagination configuration structs from Request and Application are different types and therefore cannot be passed into the same argument. Could we make it so that request.fluent.pagination and application.fluent.pagination return the same type of (optional) object? Then we could do: `pagination: self.fluent.pagination ?? self.application.fluent.pagination` here.

It could even contain an enum on how to react when the limit is exceeded:
```swift
enum MaxPerPageExceededStrategy {
    case throwError(Error)
    case clamp
}
```
Although we tend to avoid exposing enums since we cannot add new cases in the future without breaking the API.

Reply:
Both `Pagination` structs have a `fluent` property, which are different types (`Application.Fluent` and `Request.Fluent`), too, so using the same `Pagination` struct for both won't work, either.

In order to use the global limit as a fallback, I could do this: `maxPerPage: self.fluent.pagination.maxPerPage ?? self.application.fluent.pagination.maxPerPage`, but then the global limit would be used even if you specifically set the request-level limit to `nil`.

Reply:
@pulpn0ir Hmm, I see the issue. As a workaround you could do `request.fluent.pagination.maxPerPage = .max` to lift the application wide limit.

Reply:
@pulpn0ir do you think the above is a good workaround? I guess it's not so nice that it adds a useless instruction to the query (LIMIT = ....). To work around that we'd have to introduce a type that can both express a limit and the absence of a limit so we can clear the application's limit. It could be something like: `request.fluent.pagination.maxPerPage = .noLimit`. By conforming it to `ExpressibleByIntegerLiteral` we could enable `.maxPerPage = 10` but assigning from another variable would have to be something like `.maxPerPage.value = limit`.

```
.maxPerPage = nil // not set, takes application's limit if configured
.maxPerPage = 10 // overrides application's limit
.maxPerPage = .noLimit // overrides application's limit by removing it (set's maxPerPage.value to `nil`)
.maxPerPage.value = limit // overrides application's limit (allows setting from variable)
```

Reply:
This is pretty much what I did in the latest commit. Please have a look.